src/core/config/config_module/merge.rs

```diff
@@ -179,12 +179,32 @@ impl Covariant for Enum {
     }
 }
 
-impl Invariant for Cache {
-    fn unify(self, other: Self) -> Valid<Self, String> {
-        let mut types = self.config.types;
-        let mut enums = self.config.enums;
+impl Contravariant for Config {
+    fn shrink(self, _other: Self) -> Valid<Self, String> {
+        Valid::succeed(self)
+    }
+}
+
+impl Covariant for Config {
+    fn expand(self, other: Self) -> Valid<Self, String> {
+        Valid::succeed(self.merge_right(other))
```

Comment:
Actually Config can't have bivariance. We are implementing Variance only for types, interfaces, inputs and enums. They represent some structure of data, hence implementing these traits make sense. Config is a pure value and I would classify it as invariant.